Controller/Controller.php

```diff
@@ -62,25 +63,34 @@ class Controller
     private $httpCacheTime;
 
     /**
-     * @param TranslatorInterface $translator        The translator.
-     * @param Twig_Environment    $twig              The twig environment.
-     * @param TranslationFinder   $translationFinder The translation finder.
-     * @param string              $cacheDir
-     * @param boolean             $debug
-     * @param string              $localeFallback
-     * @param string              $defaultDomain
-     * @param int                 $httpCacheTime
+     * @param TranslatorInterface           $translator        The translator.
+     * @param Environment|Twig_Environment  $twig              The twig environment.
+     * @param TranslationFinder             $translationFinder The translation finder.
+     * @param string                        $cacheDir
+     * @param boolean                       $debug
+     * @param string                        $localeFallback
+     * @param string                        $defaultDomain
+     * @param int                           $httpCacheTime
+     * @throws InvalidArgumentException
      */
     public function __construct(
-        TranslatorInterface $translator,
-        Twig_Environment $twig,
+        $translator,
+        $twig,
         TranslationFinder $translationFinder,
         $cacheDir,
         $debug          = false,
         $localeFallback = '',
         $defaultDomain  = '',
         $httpCacheTime  = 86400
     ) {
+        if (!$twig instanceof \Twig_Environment && !$twig instanceof Environment) {
```

Comment:
I suggest typehinting the namespaced class name and be done with it. Namespaced class are available in 1.x and 2.x already since a long time (and Symfony 3.4 requires Twig versions having them as their min Twig version anyway).